openshift/system/Dockerfile

```diff
@@ -1,91 +1,88 @@
-FROM quay.io/centos7/ruby-27-centos7 AS porta-base
-
-USER root
-
+FROM quay.io/centos/centos:stream8 as builder
+WORKDIR /opt/system
 ENV RUBY_MAJOR_VERSION=2 \
-    RUBY_MINOR_VERSION=7
+    RUBY_MINOR_VERSION=7 \
+    RAILS_ENV=production \
+    NODE_ENV=production
+ENV RUBY_VERSION="${RUBY_MAJOR_VERSION}.${RUBY_MINOR_VERSION}"
+ADD . ./
+ADD openshift/system/config/* ./config/
+RUN dnf -y module enable ruby:${RUBY_VERSION} nodejs:16 mysql:8.0 \
+    && dnf install -y --setopt=skip_missing_names_on_install=False,tsflags=nodocs shared-mime-info make automake gcc gcc-c++ redhat-rpm-config postgresql rubygem-irb rubygem-rdoc ruby-devel nodejs libpq-devel mysql-devel gd-devel git 'dnf-command(download)' podman-catatonit \
+    && gem install bundler --version 2.2.25 \
+    && bundle config set --local without development:test:licenses \
+    && bundle config set --local deployment true \
+    && bundle install --jobs $(grep -c processor /proc/cpuinfo) --retry=5 \
+    && npm install -g yarn \
+    && yarn install:safe --no-progress \
+    && SECRET_KEY_BASE=rails/32947 bundle exec rake assets:precompile tmp:clear \
+    && rm -rf tmp/cache /usr/local/share/gems/cache /usr/local/share/gems/doc
```

Comment:
it has nothing to do with `dnf`

Reply:
The command includes a call to `dnf install`, so the downloaded packages for `shared-mime-info make automake gcc gcc-c++ redhat-rpm-config postgresql rubygem-irb rubygem-rdoc ruby-devel nodejs libpq-devel mysql-devel gd-devel git 'dnf-command(download)' podman-catatonit` must be stored in some cache folder

Reply:
Sorry, I meant that this clean-up has nothing to do with DNF cache. We don't have to clean DNF cache because this is the `builder` stage and it doesn't end up as part of the final image.